devito/types/basic.py

```diff
@@ -1014,29 +1037,31 @@ def _evaluate(self, **kwargs):
         This allow to evaluate off grid points as EvalDerivative that are better
         for the compiler.
         """
+        mapper = self._grid_map
+        subs = mapper.pop('subs', {})
         # Average values if at a location not on the Function's grid
-        if not self._grid_map:
+        if not mapper:
             return self
 
         io = self.interp_order
-        # Base function
-        if self._avg_mode == 'harmonic':
-            retval = 1 / self.function
-        else:
-            retval = self.function
+        retval = self.subs({i.subs(subs): self.indices_ref[d]
+                            for d, i in mapper.items()})
+        if 'harmonic' in self._avg_mode:
+            retval = retval.safe_inv(retval, safe='safe' in self._avg_mode)
```

Comment:
`safe='safe'` is a bit clumsy

Reply:
It's a boolean `'safe' in self._avg_mode` not `safe='safe'`